### ISSUE TITLE
Fails to build on Linux

### ISSUE THREAD

**Issue by tux3** [Terraspace/UASM#64]
Hello, I tried to build UASM on Debian Sid using the GccUnix makefile but it fails, is Linux still supported?

Here's the output I got:

``` bash
$ make -f GccUnix.mak 
gcc -D __UNIX__ -c -IH -D __UNIX__ -DNDEBUG -O2  -o GccUnixR/data.o data.c
In file included from data.c:36:0:
H/memalloc.h:49:0: warning: "alloca" redefined
  #define alloca(x)  __builtin_alloca(x)
 
In file included from /usr/include/stdlib.h:453:0,
                 from H/globals.h:38,
                 from data.c:35:
/usr/include/alloca.h:35:0: note: this is the location of the previous definition
 # define alloca(size) __builtin_alloca (size)
 
data.c:58:9: error: unknown type name ‘__int64’
 typedef __int64 INT_PTR, *PINT_PTR;
         ^~~~~~~
data.c:59:26: error: expected ‘=’, ‘,’, ‘;’, ‘asm’ or ‘__attribute__’ before ‘UINT_PTR’
 typedef unsigned __int64 UINT_PTR, *PUINT_PTR;
                          ^~~~~~~~
data.c:60:9: error: unknown type name ‘__int64’
 typedef __int64 LONG_PTR, *PLONG_PTR;
         ^~~~~~~
data.c:61:26: error: expected ‘=’, ‘,’, ‘;’, ‘asm’ or ‘__attribute__’ before ‘ULONG_PTR’
 typedef unsigned __int64 ULONG_PTR, *PULONG_PTR;
                          ^~~~~~~~~
data.c:68:8: error: unknown type name ‘UINT_PTR’
 extern UINT_PTR UTF8toWideChar(const unsigned char *pSource, UINT_PTR nSourceLen, UINT_PTR *nSourceDone, unsigned short *szTarget, UINT_PTR nTargetMax);
        ^~~~~~~~
data.c:68:62: error: unknown type name ‘UINT_PTR’; did you mean ‘PINT_PTR’?
 extern UINT_PTR UTF8toWideChar(const unsigned char *pSource, UINT_PTR nSourceLen, UINT_PTR *nSourceDone, unsigned short *szTarget, UINT_PTR nTargetMax);
                                                              ^~~~~~~~
                                                              PINT_PTR
data.c:68:83: error: unknown type name ‘UINT_PTR’; did you mean ‘PINT_PTR’?
 extern UINT_PTR UTF8toWideChar(const unsigned char *pSource, UINT_PTR nSourceLen, UINT_PTR *nSourceDone, unsigned short *szTarget, UINT_PTR nTargetMax);
                                                                                   ^~~~~~~~
                                                                                   PINT_PTR
data.c:68:132: error: unknown type name ‘UINT_PTR’; did you mean ‘PINT_PTR’?
 extern UINT_PTR UTF8toWideChar(const unsigned char *pSource, UINT_PTR nSourceLen, UINT_PTR *nSourceDone, unsigned short *szTarget, UINT_PTR nTargetMax);
                                                                                                                                    ^~~~~~~~
                                                                                                                                    PINT_PTR
data.c: In function ‘data_item’:
data.c:753:23: warning: implicit declaration of function ‘UTF8toWideChar’ [-Wimplicit-function-declaration]
                   j = UTF8toWideChar(pchar, string_len, NULL,(unsigned short *) &buff, string_len);
                       ^~~~~~~~~~~~~~
data.c:72:44: warning: implicit declaration of function ‘OutputInterleavedBytes’; did you mean ‘OutputInterleavedDataBytes’? [-Wimplicit-function-declaration]
 #define OutputInterleavedDataBytes( x, y ) OutputInterleavedBytes( x, y, NULL );
                                            ^
data.c:802:7: note: in expansion of macro ‘OutputInterleavedDataBytes’
       OutputInterleavedDataBytes(pchar, string_len);
       ^~~~~~~~~~~~~~~~~~~~~~~~~~
GccUnix.mak:34: recipe for target 'GccUnixR/data.o' failed
make: *** [GccUnixR/data.o] Error 1
```

**Comment by john-terraspace** [comment]
Hi,

 

It definitely is. There was an issue in 2.39 with the definition of those types, it’s been fixed now in 2.40 which will be up in the next day or so!

 

Cheers

John

**Comment by tux3** [comment]
Thanks! I can confirm it compiles now.
